app/controllers/images/originals_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# Images::OriginalsController
+module Images
+  class OriginalsController < ApplicationController
+    # This is an AJAX request so we can keep overhead low.
+    disable_filters
+
+    def show
+      # Because this is only called from stimulus within valid pages, we
+      # can assume the user is already logged in and the image is correct...
+      # unless a malicious actor is calling this directly, in which case we
+      # don't really care if errors are handled gracefully.
+      @user = User.current = session_user or raise("login required")
+      @image = Image.safe_find(params[:id]) or raise("image not found")
+      log_request(@user, @image)
+
+      respond_to do |format|
+        format.html { redirect_to(@image.cached_original_url) }
```

Comment:
I think another way to avoid the complaint about this line would be to create a route for it that generates the right URL - something like `local_original_image` and then use `local_orginal_image_path(@image.id)`.

Reply:
Interesting, I didn't know routes could do that, will look into...

Reply:
I'm struggling with this.  This is close:

```
resources :images, only: [:show] do
  member do
    get("cached_original", to: static("/orig_cache/#{id}.jpg")
  end
end
```

But I can't figure out how to get the id in there.  (And ensure that id is an integer!  Otherwise our security "fix" ends up making it *less* secure!)  And recommendations?